### PR TITLE
[addonservices] allow offline mode

### LINKED ISSUE #2628: [marketplace] Expecting no access to the marketplace when remote is set to false
For users who decided to rely only on an offline distribution and so who setup openHAB like this:
```

# Access Remote Add-on Repository
# Defines whether the remote openHAB add-on repository should be used for browsing and installing add-ons. (default is true)
#
remote = false
```

I assume MainUI should not show marketplace in this case but only bindings present in the kar file.

The current behaviour is to have bindings from the marketplace displayed in UI.

### PR DESCRIPTION
**Pull request by J-N-K**
Fixes #2628 

Signed-off-by: Jan N. Klug <jan.n.klug@rub.de>

### CODE REVIEW

**Comment by J-N-K** [comment]
Since there is a lot of common code between both addon services, it might be a good idea to refactor them to inherit from an `AbstractAddonService`. But this is out of scope here and should probably be done after a solution for #2607 is found.

**Comment by J-N-K** [comment]
Test fails unrelated.

**Review by cweitkamp — COMMENTED** [review]
Thank you for looking into this. From my pov we now reached some point where we need unit tests to cover the functionality of those addon services. Wdyt?
If you like I can give you a hand on that. Just ping me.

And I agree. The code can be reduced by introducing an abstract base class.

**Comment by J-N-K** [comment]
Regarding the unit tests: I agree. Unfortunately it's not that easy to mock the remote end (and it's also different for both add-on services. 

This will improve with the introduction of the `AbstractAddonService` as a `TestAddonService` could at least test the common parts. 

IMO the bundle installers also need test coverage. I can have a look in a separate PR.

**Review by cweitkamp — APPROVED** [review]
LGTM. Thanks.

🎆 Happy New Year.

File: bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/community/CommunityMarketplaceAddonService.java [hunk 1]
```diff
@@ -460,4 +475,13 @@ private void postFailureEvent(String extensionId, @Nullable String msg) {
         Event event = AddonEventFactory.createAddonFailureEvent(extensionId, msg);
         eventPublisher.post(event);
     }
+
+    private boolean remoteEnabled() {
+        try {
+            Configuration configuration = configurationAdmin.getConfiguration("org.openhab.addons", null);
+            return (boolean) Objects.requireNonNullElse(configuration.getProperties().get("remote"), true);
```
Comment (cweitkamp) [inline bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/community/CommunityMarketplaceAddonService.java:482]: It looks.like the [getProperties()](https://docs.osgi.org/javadoc/osgi.cmpn/7.0.0/org/osgi/service/cm/Configuration.html#getProperties--) can return `null` and we are potentially run into NPEs here.

Comment (J-N-K) [reply bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/community/CommunityMarketplaceAddonService.java:482]: Fixed in https://github.com/openhab/openhab-core/pull/2607/commits/fc25ddbf41f4671b165ec795d537e5a6adb1e35a

Comment (cweitkamp) [reply bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/community/CommunityMarketplaceAddonService.java:482]: Thank you.

I will try to find a solution for #2581 later.

**Pull request merged**